Server/ConcordiaCurriculumManager/Services/DossierReviewService.cs

```diff
@@ -117,10 +130,48 @@ private async Task AcceptDossierChanges(Dossier dossier)
             _logger.LogError($"Encountered error attempting to accept dossier {dossier.Id} and remove it from the review process");
     }
 
+    public async Task<Dossier> GetDossierWithDiscussion(Guid dossierId)
+    {
+        var dossier = await GetDossierWithApprovalStagesAndRequestsAndDiscussionOrThrow(dossierId);
+
+        if (dossier.State.Equals(DossierStateEnum.Created))
+        {
+            throw new BadRequestException("The dossier is not published yet.");
+        }
+
+        return dossier;
+    }
+
+    public async Task AddDossierDiscussionReview(Guid dossierId, DiscussionMessage message)
+    {
+        var userId = _httpContextAccessor.HttpContext?.User.Claims?.FirstOrDefault(c => c.Type.Equals(Claims.Id)) ?? throw new BadRequestException("User Id does not exist");
```

Comment:
There's a method `GetCurrentUserClaim` in UserAuthenticationService to get this, so we don't duplicate code, and then we can remove the reference to httpContextAccessor from the constructor.